### ISSUE TITLE
When tapped inside the Webview, Flutter-Window loses focus

### ISSUE THREAD

**Issue by Eerey** [jnschulze/flutter-webview-windows#230]
If a `Webview`-widget is used and the user taps (any pointer device/mouse/onTapDown) within the `Webview`, the flutter-window loses it's focus.

This is not desired because in our project we want to use a customized `Virtual Keyboard` that utilizes `TapRegion` in order to map the `Virtual Keyboard` to the text input controls of the `Webview`

more on `TapRegion`:
https://github.com/flutter/flutter/pull/107262

**Comment by paderick** [comment]
same here

**Comment by NickYoung-max** [comment]
have you get a solution yet?

**Comment by Eerey** [comment]
> have you get a solution yet?

No, sadly. Currently we use an external Virtual Keyboard that is not part of the main flutter app.

**Comment by mkilling** [comment]
@huycn Does your change work to fix this issue? Is there any more work required?

**Comment by huycn** [comment]
> @huycn Does your change work to fix this issue? Is there any more work required?

@mkilling No, it was not a proper fix. It created other issue. If I remember it well, with that commit, the focus appears to stay in parent window, but it's not, and no other widgets can get the focus again, until main window is minimized and restored again.

Actually, it would give the same result if I pass flutter's hWnd (from `registrar->GetView()->GetNativeWindow()`) as `hWndParent` to `CreateWindowEx` here https://github.com/jnschulze/flutter-webview-windows/blob/main/windows/webview_windows_plugin.cc#L196 instead of `nullptr`.

Anyway, it didn't work hence I reverted commit. I ended up with a workaround using window_manager plugin to take back focus after the click.

**Comment by mkilling** [comment]
@jnschulze We are considering trying to provide a fix for this. Do you have any hints for us that could get us started? E.g. ideas what might be causing the loss of focus, or potential solution ideas?

**Comment by omar-hanafy** [comment]
Thanks for the detailed reports above. We’re seeing the same behavior in our Flutter desktop app: every time the WebView2 surface receives the first mouse down, the parent Flutter window loses activation (title bar grays out) and keyboard shortcuts stop working until a second click re-focuses the window.

Has there been any movement on this? If there’s a known workaround—e.g., handling WM_MOUSEACTIVATE or forcing SetActiveWindow on the parent HWND.

**Comment by omar-hanafy** [comment]
Hi @Eerey, did find any workaround for this issue? since it was opened in 2023

**Comment by Eerey** [comment]
Hi @omar-hanafy Yes, the current workaround is that we fully implemented a customized On-Screen-Keyboard (in Flutter, separate instance/project) that never receives the focus. See below to see the window settings that are set at runtime.

In order to position/set the OSK correctly, I established start parameters that gets injected by the main program with the WebView.

I don't like this solution at all, but it solves our issue for now.


```dart
// window settings for the On-Screen-Keyboard, not the main App with the WebView

// find HWND
final hwnd = FindWindow(
        'FLUTTER_RUNNER_WIN32_WINDOW'.toNativeUtf16(), guid.toNativeUtf16());

SetWindowLongPtr(hwnd, WINDOW_LONG_PTR_INDEX.GWL_EXSTYLE,
    WINDOW_EX_STYLE.WS_EX_NOACTIVATE);
await windowManager.ensureInitialized();
await windowManager.setAlwaysOnTop(true);
```

**Comment by omar-hanafy** [comment]
@Eerey 

I published a maintained fork/rebuild of this plugin here:

Package: https://pub.dev/packages/webview_flutter_windows
Repo: https://github.com/omar-hanafy/webview_flutter_windows

v1.0.0 includes Flutter/Dart modernization, updated Windows/WebView2 integration, focus and text-input fixes, cookie APIs, lifecycle/dispose stability fixes, and updated
docs/tests/example app.

I'm happy to open a draft PR for v1 if there's interest.

cc @jnschulze

**Comment by jnschulze** [comment]
@omar-hanafy 
Great work!
I took a brief look at your fork. It looks like you've put a lot of effort into it, so thank you for that.

I haven't had as much time to dedicate to the project as I'd have liked over the past few years, but I'm hoping to give it some more attention again and put a bit more love into it going forward.

If you'd be interested in contributing some of your changes back upstream, I'd be happy to review them.
To make review and discussion easier, it would help if you could split the work into separate pull requests, with each PR focused on a single issue ( or feature).
I'd also appreciate your perspective on some of the existing open pull requests. If you have a chance to look through them, it would be helpful to know whether any of them already address issues that are currently still marked as open, or whether there are overlaps with work you've already done.
That kind of feedback would help me get a clearer picture of the current state of the project and prioritize reviews.

**Comment by omar-hanafy** [comment]
@jnschulze 

Thanks! glad you're picking this back up, and I appreciate the offer to review.

Agreed that focused, single-purpose PRs are the right shape. Realistically though, I probably will not have time over the next few months to split the fork properly, and I do not want to imply I can send clean PRs in the next few days. When I do get time, I would likely start with the focus fix relevant to this issue (#230).

That said, if it is easier on your side to look at the fork as a new baseline, or even as one larger PR, I am open to whichever path creates the least review work for you.

On the open PRs: I recently went through them while deciding what to bring into the fork. The ones that look worth prioritizing are #325, #328, #321, #327, plus #276 as a useful addition. The cookie PRs (#299/#256/#221) and #324/#178 look overlapping or already covered.
